File: web/src/api/promotion.ts
import http from './http'

export function pagePromotions(params: { page?: number; size?: number }) {
  return http.get<any>('/promotions', { params })
}

export function listAllPromotions() {
  return http.get<any>('/promotions/all')
}

export function createPromotion(data: {
  promoCode: string
  promoName: string
  promoType: string
  thresholdCent: number
  discountAmountCent: number
  beginAt?: string
  endAt?: string
  status?: number
}) {
  return http.post<any>('/promotions', data)
}

export function updatePromotion(
  id: number,
  data: {
    promoName?: string
    promoType?: string
    thresholdCent?: number
    discountAmountCent?: number
    beginAt?: string
    endAt?: string
    status?: number
  }
) {
  return http.put<any>(`/promotions/${id}`, data)
}

File: src/main/java/com/orderflow/domain/entity/Promotion.java
package com.orderflow.domain.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("promotion")
public class Promotion {
    @TableId(type = IdType.AUTO)
    private Long id;
    private Long tenantId;
    private String promoCode;
    private String promoName;
    private String promoType;
    private Long thresholdCent;
    private Long discountAmountCent;
    private LocalDateTime beginAt;
    private LocalDateTime endAt;
    private Integer status;
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}
